src/bootstrap/channel.rs

```diff
@@ -12,11 +12,13 @@ use build_helper::output;
 
 use crate::Build;
 
-pub struct GitInfo {
-    inner: Option<Info>,
+pub enum GitInfo {
+    Ignore,
+    Missing,
+    Present(Info),
```

Comment:
Can we add comments to these variants explaining their meaning? (In particular, Ignore not indicating no information, but rather positive presence, is not obvious).

Alternatively restructuring to Missing and a `Present(Option<Info>)` variant might be better; it would reduce the explaining needed I think.

Reply:
> Can we add comments to these variants explaining their meaning? (In particular, Ignore not indicating no information, but rather positive presence, is not obvious).

Sure, done.

> Alternatively restructuring to Missing and a Present(Option<Info>) variant might be better; it would reduce the explaining needed I think.

Hmm, this doesn't seem any more clear to me - you'd still need a comment explaining what `Present(None)` means.

Reply:
Ignore doesn't make sense in the context of "what is the git info" - it's an outside parameter. Present(None) does need a comment, I agree, but is still imo clearer in terms of present (but no info). Mainly because if I see a match in (future) code for Present, I won't likely think to check for Absent, but this way the check will be more obviously necessary.

Reply:
:+1: Makes sense, I changed it.